iocage_lib/ioc_json.py

```diff
@@ -44,19 +44,672 @@
+
+        # Version 2 keys
+        if not conf.get('sysvmsg'):
+            conf['sysvmsg'] = 'new'
+        if not conf.get('sysvsem'):
+            conf['sysvsem'] = 'new'
+        if not conf.get('sysvshm'):
+            conf['sysvshm'] = 'new'
+
+        # Version 4 keys
+        if not conf.get('basejail'):
+            conf['basejail'] = 'no'
+
+        # Version 5 keys
+        if not conf.get('comment'):
+            conf['comment'] = 'none'
+
+        # Version 6 keys
+        if not conf.get('host_time'):
+            conf['host_time'] = 'yes'
+
+        # Version 7 keys
+        if not conf.get('depends'):
+            conf['depends'] = 'none'
+
+        # Version 9 keys
+        if not conf.get('dhcp'):
+            conf['dhcp'] = 'off'
+        if not conf.get('bpf'):
+            conf['bpf'] = 'no'
+
+        # Version 10 keys
+        if not conf.get('vnet_interfaces'):
+            conf['vnet_interfaces'] = 'none'
+
+        # Version 11 keys
+        if not conf.get('hostid_strict_check'):
+            conf['hostid_strict_check'] = 'off'
+
+        # Version 12 keys
+        if not conf.get('allow_mlock'):
+            conf['allow_mlock'] = '0'
+
+        # Version 13 keys
+        if not conf.get('vnet_default_interface'):
+            conf['vnet_default_interface'] = 'none'
+
+        # Version 14 keys
+        if not conf.get('allow_tun'):
+            conf['allow_tun'] = '0'
+
+        # Version 15 keys
+        if not conf.get('allow_mount_fusefs'):
+            conf['allow_mount_fusefs'] = '0'
+
+        if not default:
+            conf.update(jail_conf)
+
+        return conf, True
+
+    def check_jail_config(self, conf):
+        """
+        Checks the jails configuration and migrates anything needed
+        """
+        release = conf.get('release', None)
+        template = conf.get('template', 'no')
+        renamed = False
+
+        if release is None:
+            err_name = self.location.rsplit('/', 1)[-1]
+            iocage_lib.ioc_common.logit(
+                {
+                    'level': 'EXCEPTION',
+                    'message': f'{err_name} has a corrupt configuration,'
+                               ' please destroy the jail.'
+                },
+                _callback=self.callback,
+                silent=self.silent)
+
+        release = release.rsplit('-p', 1)[0]
+        cloned_release = conf.get('cloned_release', 'LEGACY_JAIL')
+
+        try:
+            freebsd_version = f'{self.iocroot}/releases/{release}/root/bin/' \
+                'freebsd-version'
+        except FileNotFoundError:
+            if template == 'yes':
+                freebsd_version = f'{self.iocroot}/templates/' \
+                    f"{conf['host_hostuuid']}/root/bin/freebsd-version"
+            else:
+                temp_uuid = self.location.rsplit('/', 1)[-1]
+                freebsd_version = f'{self.iocroot}/jails/{temp_uuid}' \
+                    '/root/bin/freebsd-version'
+        except KeyError:
```

Comment:
I am not sure where a `KeyError` could be raised here ?

Reply:
If we are looking at key `host_hostuuid` of `conf`, I think the `KeyError` catch clause will have to be moved under `FileNotFoundError` as the current one would not cover it